packages/core/src/components/entity-title/entityTitle.tsx

```diff
@@ -0,0 +1,148 @@
+/* !
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ */
+
+import classNames from "classnames";
+import * as React from "react";
+
+import { IconName, IconNames } from "@blueprintjs/icons";
+
+import { Classes, DISPLAYNAME_PREFIX, MaybeElement, Props } from "../../common";
+import { H1, H2, H3, H4, H5, H6 } from "../html/html";
+import { Icon } from "../icon/icon";
+import { Text } from "../text/text";
+
+export enum HeadingSize {
+    h6 = "h6",
+    h5 = "h5",
+    h4 = "h4",
+    h3 = "h3",
+    h2 = "h2",
+    h1 = "h1",
+}
+
+export type IEntityTitleProps = Props & {
+    title: JSX.Element | string;
+
+    /**
+     * Inherit font size if not provided
+     */
+    headingSize?: HeadingSize;
+
+    titleURL?: string;
+
+    /**
+     * Name of a Blueprint UI icon (or an icon element) to render in the section's header.
+     * Note that the header will only be rendered if `title` is provided.
+     */
+    icon?: IconName | MaybeElement;
+
+    /**
+     * <Tag> components work best - if multiple, wrap in <React.Fragment>
+     */
+    tags?: React.ReactNode;
+
+    subtitle?: JSX.Element | string;
+};
+
+export const EntityTitle: React.FC<IEntityTitleProps> = ({
+    className,
+    headingSize,
+    title,
+    icon,
+    tags,
+    subtitle,
+    titleURL,
+}) => {
+    const titleComponent = React.useMemo(() => {
+        const maybeTitleWithURL =
+            titleURL != null ? (
+                <a target="_blank" href={titleURL} rel="noreferrer">
+                    {title}
+                </a>
+            ) : (
+                title
+            );
+
+        const titleClassName = classNames(Classes.ENTITY_TITLE_TITLE);
+
+        switch (headingSize) {
```

Comment:
instead of this switch case syntax and exposing a new `HeadingSize` enum in the public API, I think it might be nicer to have a prop that accepts a function component:

```tsx
<EntityTitle title="Foo" heading={H4} />

interface EntityTitleProps {
    /**
     * React component to render the main title heading. This defaults to Blueprint's `<Text>` component,
     * which inherits font size from its containing element(s).
     *
     * To render larger, more prominent titles, Use Blueprint's heading components instead
     * (e.g. `{ H1 } from "@blueprintjs/core"`).
     *
     * @default Text
     */
    heading?: React.FC<any>;
}
```

then you can simply render:

```ts
React.createElement(heading, { className: Classes.ENTITY_TITLE_TITLE }, maybeTitleWithURL);
```